docs/sources/installation/binaries.md

```diff
@@ -32,18 +32,52 @@ runtime:
 Docker in daemon mode has specific kernel requirements. For details,
 check your distribution in [*Installation*](../#installation-list).
 
-In general, a 3.8 Linux kernel is the minimum requirement for Docker, as
-some of the prior versions have known issues that are triggered by Docker.
-Linux kernel versions older than 3.8 are known to cause kernel panics and
-to break Docker.
+A 3.10 Linux kernel is the minimum requirement for Docker.
+Kernels older than 3.10 lack some of the features required to run Docker
+containers. These older versions are known to have bugs which cause data loss
+and frequently panic under certain conditions.
+We recommend kernel 3.10 or newer to be used.
 
 The latest minor version (3.x.y) of the 3.10 (or a newer maintained version)
 Linux kernel is recommended. Keeping the kernel up to date with the latest
 minor version will ensure critical kernel bugs get fixed.
 
+> **Warning**:
+> Installing custom kernels and kernel packages is likely to not be
+> supported by your Linux distribution's vendor. Please make sure to
+> ask your vendor about Docker support first before attempting to
+> install custom kernels on your distribution.
+
+> **Warning**:
+> Installing a newer kernel might not be enough for some distributions
+> which provide packages which are too old or incompatible with
+> newer kernels.
+
 Note that Docker also has a client mode, which can run on virtually any
 Linux kernel (it even builds on OS X!).
 
+## Check if AppArmor and SELinux are enabled
+
+Some Linux distributions enable AppArmor or SELinux by default and
+they run a kernel which doesn't meet the minimum requirements (3.10
+or newer). Updating the kernel to 3.10 or newer on such a system
+might not be enough to start Docker and run containers.
+Incompatibilities between the version of AppArmor/SELinux user
+space utilities provided by the system and the kernel could prevent
+Docker from running, from starting containers or make containers
+exhibit unexpected behaviour.
+
+> **Warning**:
+> If any of the two security mechanisms is enabled, it should not be
+> disabled to make Docker or its containers run. This will reduce
+> security in that environment, lose support from the distribution's
+> vendor for the system and might break regulations, and security
+> policies in heavily regulated environments.
+
+> **Warning**:
```

Comment:
reducing the security of your system will reduce its ability to prevent data loss - but it kinda feels to me like this `Warning` really should be the first paragraph - so I'll try it there in my version of this PR